numpy/linalg/linalg.py

```diff
@@ -2201,18 +2201,7 @@ def norm(x, ord=None, axis=None, keepdims=False):
                 ord + 1
             except TypeError:
                 raise ValueError("Invalid norm order for vectors.")
-            if x.dtype.type is longdouble:
-                # Convert to a float type, so integer arrays give
-                # float results.  Don't apply asfarray to longdouble arrays,
-                # because it will downcast to float64.
-                absx = abs(x)
-            else:
-                absx = x if isComplexType(x.dtype.type) else asfarray(x)
-                if absx.dtype is x.dtype:
-                    absx = abs(absx)
-                else:
-                    # if the type changed, we can safely overwrite absx
-                    abs(absx, out=absx)
+            absx = abs(x)
```

Comment:
The comment stated this was done for integer types. Without conversion, those now would seem to have a bigger risk of overflow in the `.reduce` below. I guess converting to float is not unreasonable since the `**(1/ord)` at the end will do that anyway. On the other hand, I prefer the simplicity of this change, and it is also what is done above for `ord=2` (surely, the most common case).